state/subnets.go

```diff
@@ -230,6 +232,88 @@ func (s *Subnet) Refresh() error {
 	return nil
 }
 
+// AdditiveUpdate adds new info to the subnet based on provided info.  There are
+// restrictions:
+//   - no change to CIDR, more work to determine how to handle needs to
+// be done.
+//   - no change to ProviderId nor ProviderNetworkID, these are immutable.
+func (s *Subnet) AdditiveUpdate(args SubnetInfo) error {
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt != 0 {
+			if err := s.Refresh(); err != nil {
+				if errors.IsNotFound(err) {
+					return nil, errors.Errorf("ProviderId %q not unique", args.ProviderId)
+				}
+				return nil, errors.Trace(err)
+			}
+		}
+		var ops []txn.Op
+		makeSpaceNameUpdate, err := s.updateSpaceName(args.SpaceName)
+		if err != nil {
+			return nil, err
+		}
+		if makeSpaceNameUpdate {
+			ops = append(ops, txn.Op{
+				C:      subnetsC,
+				Id:     s.doc.DocID,
+				Assert: bson.D{{"txn-revno", s.doc.TxnRevno}},
+				Update: bson.D{{"$set", bson.D{{"space-name", args.SpaceName}}}},
```

Comment:
It is possible to accrue the `bson.DocElems` that go in the `bson.D` (one op) versus an op for each field being set.

Reply:
done